assets/js/blocks/cart-checkout-shared/use-forced-layout.ts

```diff
@@ -41,6 +42,8 @@ export const useForcedLayout = ( {
 } ): void => {
 	const currentRegisteredBlocks = useRef( registeredBlocks );
 	const currentDefaultTemplate = useRef( defaultTemplate );
+	const [ forcedBlocksInserted, setForcedBlocksInserted ] =
```

Comment:
Keeping a count of inserted blocks feels a bit hacky here but I'm struggling to understand what we're trying to solve, why do we need to refresh the selectors after each block insert?

Reply:
Because this line: https://github.com/woocommerce/woocommerce-blocks/blob/587d04f0619faa8e07997d20f6d5c745f09283d7/assets/js/blocks/cart-checkout-shared/use-forced-layout.ts#L107 is used to determine whether we need to insert the inner block again. Since the selector isn't re-run,  `innerBlocks` is still empty, despite it actually being updated in the previous run of this hook.